### PR TITLE
Fix COFF reloc naming

### PR DESCRIPTION
**Pull request by Roeegg2**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Not very big changes, explanations in commit descriptions

...

**Test plan**

All tests green
...

**Closing issues**

N/A
...

### CODE REVIEW

File: librz/bin/format/coff/coff_reloc.c [hunk 1]
```diff
@@ -97,66 +134,42 @@ static void relocs_foreach(struct rz_bin_coff_obj *bin, RelocsForeachCb cb, void
 			ut8 patch_buf[8];
 			if (sym_vaddr) {
 				switch (bin->hdr.f_magic) {
+				// TODO: Missing handling of MIPS architecture
 				case COFF_FILE_MACHINE_I386:
 					switch (rel[j].rz_type) {
 					case COFF_REL_I386_DIR32:
```
Comment (wargio) [inline librz/bin/format/coff/coff_reloc.c]: create a function also for this.

File: librz/bin/format/coff/coff_reloc.c [hunk 2]
```diff
@@ -35,6 +35,43 @@ RZ_API ut64 rz_coff_import_index_addr(struct rz_bin_coff_obj *obj, ut64 imp_inde
 
 typedef void (*RelocsForeachCb)(RZ_BORROW RzBinReloc *reloc, ut8 *patch_buf, size_t patch_buf_sz, void *user);
 
+static size_t reloc_general_arch_rel32_common(struct rz_bin_coff_obj *bin, struct coff_reloc *rel, RzBinReloc *reloc, ut64 sym_vaddr, ut8 *patch_buf) {
```
Comment (wargio) [inline librz/bin/format/coff/coff_reloc.c]: what about adding `name` as argument which is used to set `print_name`, so you don't have to specify name afterwards?

File: librz/bin/format/coff/coff_reloc.c [hunk 3]
```diff
@@ -35,6 +35,43 @@ RZ_API ut64 rz_coff_import_index_addr(struct rz_bin_coff_obj *obj, ut64 imp_inde
...
+		return 0;
+	}
+	reloc->addend = data;
+	data += sym_vaddr - reloc->vaddr - 4;
+	rz_write_le32(patch_buf, (st32)data);
+
+	return 4;
+}
+
+static size_t reloc_arm_branches_common(struct rz_bin_coff_obj *bin, struct coff_reloc *rel, RzBinReloc *reloc, ut64 sym_vaddr, ut8 *patch_buf) {
```
Comment (wargio) [inline librz/bin/format/coff/coff_reloc.c]: same here?

File: librz/bin/format/coff/coff_reloc.c (hunk 1 above)
Comment (Roeegg2) [reply librz/bin/format/coff/coff_reloc.c]: a function for the whole switch statement?

Comment (wargio) [reply librz/bin/format/coff/coff_reloc.c]: Yes, for the whole `switch (rel[j].rz_type)`

**Comment by Roeegg2** [comment]
@wargio fixed everything, let me know what you think

File: librz/bin/format/coff/coff_reloc.c [hunk 4]
```diff
@@ -35,6 +35,100 @@ RZ_API ut64 rz_coff_import_index_addr(struct rz_bin_coff_obj *obj, ut64 imp_inde
...
+	switch (reloc_type) {
+	case COFF_REL_I386_DIR32:
+		reloc->type = RZ_BIN_RELOC_32;
+		rz_write_le32(patch_buf, (ut32)sym_vaddr);
+		reloc->print_name = "IMAGE_REL_I386_32";
+		return 4;
+	case COFF_REL_I386_REL32:
+		return reloc_general_arch_rel32_common(bin, reloc, sym_vaddr, patch_buf, "IMAGE_REL_I386_REL32");
+		// TODO: Missing handling of other relocation types
+	}
```
Comment (wargio) [inline librz/bin/format/coff/coff_reloc.c:89]:
```suggestion
	default:
		break;
	}
```

File: librz/bin/format/coff/coff_reloc.c [hunk 5]
```diff
@@ -35,6 +35,100 @@ RZ_API ut64 rz_coff_import_index_addr(struct rz_bin_coff_obj *obj, ut64 imp_inde
...
+
+	return 0;
+}
+
+static ut8 handle_amd64_relocs(struct rz_bin_coff_obj *bin, RzBinReloc *reloc, ut64 sym_vaddr, ut8 *patch_buf, ut16 reloc_type) {
+	switch (reloc_type) {
+	case COFF_REL_AMD64_REL32:
+		return reloc_general_arch_rel32_common(bin, reloc, sym_vaddr, patch_buf, "IMAGE_REL_AMD64_REL32");
+		// TODO: Missing handling of other relocation types
+	}
```
Comment (wargio) [inline librz/bin/format/coff/coff_reloc.c:102]:
```suggestion
	default:
		break;
	}
```

File: librz/bin/format/coff/coff_reloc.c [hunk 6]
```diff
@@ -35,6 +35,100 @@ RZ_API ut64 rz_coff_import_index_addr(struct rz_bin_coff_obj *obj, ut64 imp_inde
...
+}
+
+static ut8 handle_arm_relocs(struct rz_bin_coff_obj *bin, RzBinReloc *reloc, ut64 sym_vaddr, ut8 *patch_buf, ut16 reloc_type) {
+	switch (reloc_type) {
+	case COFF_REL_ARM_BRANCH24T:
+		return reloc_arm_branches_common(bin, reloc, sym_vaddr, patch_buf, "IMAGE_REL_ARM_BRANCH24T");
+	case COFF_REL_ARM_BLX23T:
+		return reloc_arm_branches_common(bin, reloc, sym_vaddr, patch_buf, "IMAGE_REL_ARM_BLX23T");
+		// TODO: Missing handling of other relocation types
+	}
```
Comment (wargio) [inline librz/bin/format/coff/coff_reloc.c:117]:
```suggestion
	default:
		break;
	}
```

File: librz/bin/format/coff/coff_reloc.c [hunk 7]
```diff
@@ -35,6 +35,100 @@ RZ_API ut64 rz_coff_import_index_addr(struct rz_bin_coff_obj *obj, ut64 imp_inde
...
+		if (!rz_buf_read_le32_at(bin->b, reloc->paddr, &data)) {
+			break;
+		}
+		ut64 dst = sym_vaddr - reloc->vaddr;
+		data |= (ut32)((dst >> 2) & 0x3ffffffULL);
+		rz_write_le32(patch_buf, data);
+		reloc->print_name = "IMAGE_REL_ARM64_BRANCH26";
+		return 4;
+		// TODO: Missing handling of other relocation types
+	}
```
Comment (wargio) [inline librz/bin/format/coff/coff_reloc.c:139]:
```suggestion
	default:
		break;
	}
```

**Review by wargio — CHANGES_REQUESTED** [review]
just add the default case in all the switches and can be merged.

**Comment by Roeegg2** [comment]
fixed.
I also added a warnings if we reach `default`

**Review by XVilka — APPROVED** [review]

**Pull request merged**